### PR TITLE
Update solution

### PR DESCRIPTION
**Pull request by bernardlee**
Update solution for cohort 016.

### CODE REVIEW

File: lib/challenge.rb [hunk 1]
```diff
@@ -16,21 +21,12 @@ def each
       next_fib = current_fib + previous_fib
     end
   end
-
-  attr_reader :stop
```
Comment (jrhorn424) [inline lib/challenge.rb]: It's not a problem that you re-arranged these blocks. But when I write Comparable or Enumerable classes, I personally prefer `<=>` or `each` to be defined before `initialize`.

File: spec/challenge_spec.rb [hunk 2]
```diff
@@ -17,8 +20,6 @@
 
   describe '#each' do
     it 'stores minimal state' do
-      require 'binding_of_caller'
```
Comment (jrhorn424) [inline spec/challenge_spec.rb]: Any particular reason moving dependencies away from their point of injection is preferable?

If this were a class, I could be persuaded. This is a test. A spec file can get very long.

File: lib/challenge.rb [hunk 3]
```diff
@@ -16,21 +21,12 @@ def each
...
-  private :stop
```
Comment (jrhorn424) [inline lib/challenge.rb]: Any reason `stop` is no longer private?

File: lib/challenge.rb (hunk 1 above)
Comment (bernardlee) [inline lib/challenge.rb]: I just wrote it in this order out of habit (modules, attributes, constructor, and then everything else).

File: lib/challenge.rb (hunk 3 above)
Comment (bernardlee) [inline lib/challenge.rb]: I didn't see it as a requirement and it didn't seem necessary as long as `@stop` is read-only.

File: spec/challenge_spec.rb (hunk 2 above)
Comment (bernardlee) [inline spec/challenge_spec.rb]: Similar to the comment about rearranging the body of a class, it was just a habit of placing `require*`s at the top of a file.

File: lib/challenge.rb (hunk 3 above)
Comment (jrhorn424) [inline lib/challenge.rb]: If someone naively uses `stop`, and it's private, it errors. That tells the person "don't do that".

Someone can introspect an instance, see a stop method if its public, and start playing with it. Not a problem here, but you can see why not considering whether something should be private or not could be a problem.

That's why I choose personally to default to making it private until I know how I want the API exposed.

**Pull request closed without merging**